src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -468,5 +468,11 @@ cdef class AioServer:
         If the Cython representation is deallocated without underlying objects
         freed, raise an RuntimeError.
         """
+        # TODO(lidiz) if users create server, and then dealloc it immediately.
+        # There is a potential memory leak of created Core server.
         if self._status != AIO_SERVER_STATUS_STOPPED:
-            raise RuntimeError('__dealloc__ called on running server: %d', self._status)
+            _LOGGER.warn(
```

Comment:
`_LOGGER.warning()`?

Reply:
If `__dealloc__` is being called on a running server, we have a logic issue. `__del__` should be called before `__dealloc__`. Can we shut the server down there?

Reply:
Fixed the `warning` typo.

That is what I'am expecting that `__del__` is called before `__dealloc__`, however, this line is reached during the development of this PR. The repro condition is simply letting server go out-of-scope without invoking `stop`.

We already invoked shutdown in the Python level `__del__`. Calling shutdown again in `__dealloc__` may resurrect the object, or lead to undefined behavior.